flixel/addons/display/FlxClothSprite.hx

```diff
@@ -0,0 +1,517 @@
+	 * @param	Rows			Number of rows of the created mesh.
+	 * @param	pinSide			The pinned side that points are not affected by wind or velocity. Use FlxObject.UP, DOWN, LEFT, RIGHT, NONE, ANY, etc.
+	 */
+	public function new(?X:Float = 0, ?Y:Float = 0, ?SimpleGraphic:FlxGraphicAsset, ?Columns:Int = 0, ?Rows:Int = 0, pinSide:Int = FlxObject.UP, crossingConstraints:Bool = false) 
+	{
+		super(X, Y, SimpleGraphic);
+		
+		this.pinSide = pinSide;
+		this.rows = Std.int(Math.max(2, Rows));
+		this.columns = Std.int(Math.max(2, Columns));
+		this.crossingConstraints = crossingConstraints;
+		
+		_drawOffset = new Point();
+		setMesh(columns, rows);
+	}
+	
+	/**
+	 * WARNING: This will remove this sprite entirely. Use kill() if you want to disable it temporarily only and reset() it later to revive it.
+	 * Used to clean up memory.
+	 */
+	override public function destroy():Void
+	{
+		points = null;
+		constraints = null;
+		_vertices = null;
+		_indices = null;
+		_uvtData = null;
+		colors = null;
+		_drawOffset = null;
+		
+		meshVelocity = FlxDestroyUtil.put(meshVelocity);
+		meshFriction = FlxDestroyUtil.put(meshFriction);
+		_meshPixels = FlxDestroyUtil.dispose(_meshPixels);
+		
+		super.destroy();
+	}
+	
+	/**
+	 * Core update loop
+	 */
+	override public function update(elapsed:Float):Void
+	{
+		updatePoints(elapsed);
+		for (i in 0...iterations) 
+		{
+			updateConstraints(elapsed);
+		}
+		
+		super.update(elapsed);
+	}
+	
+	/**
+	 * Called by game loop, updates then blits or renders current frame of animation to the screen
+	 */
+	override public function draw():Void 
+	{
+		if (_frame == null || _meshPixels == null)
+		{
+			super.draw();
+			return;
+		}
+		
+		if (alpha == 0 || _frame.type == FlxFrameType.EMPTY)
+		{
+			return;
+		}
+		
+		if (dirty)	//rarely 
+		{
+			calcFrame();
+		}
+		
+		calcImage();
+		drawImage();
+		
+		for (camera in cameras)
+		{
+			if (!camera.visible || !camera.exists || !isOnScreen(camera))
+			{
+				continue;
+			}
+			
+			getScreenPosition(_point, camera).subtractPoint(offset);
+			
+			var cr:Float = colorTransform.redMultiplier;
+			var cg:Float = colorTransform.greenMultiplier;
+			var cb:Float = colorTransform.blueMultiplier;
+			
+			var simple:Bool = isSimpleRender(camera);
+			if (simple)
+			{
+				if (isPixelPerfectRender(camera))
+				{
+					_point.floor();
+				}
+				
+				_flashPoint = new Point(_point.x + _drawOffset.x, _point.y + _drawOffset.y);
+				camera.copyPixels(_frame, _meshPixels, _meshPixels.rect, _flashPoint, cr, cg, cb, alpha, blend, antialiasing);
+			}
+			else
+			{
+				_point.add(_drawOffset.x, _drawOffset.y);
+				if (isPixelPerfectRender(camera))
+				{
+					_point.floor();
+				}
+				
+				// Create a temporary frame and draw mesh's bitmapData
+				var frameGraphic:FlxGraphic = FlxGraphic.fromBitmapData(framePixels, true, null, false);
+				
+				camera.drawTriangles(frameGraphic, _vertices, _indices, _uvtData, colors, _point, blend, antialiasing);
```

Comment:
@Beeblerox 
I'm using `camera.drawTriangles()` here the same way is used on `FlxStrip`. I didn't extended it because I prefer keep _vertices, _incides and _uvtData private, and I would rewrite almost everything on `draw()` method.
For flash `camera.drawTriangles()` is slow. So I still using `camera.drawPixels()`.